app/controllers/api/generic_object_definitions_controller.rb

```diff
@@ -136,17 +137,11 @@ def generic_objects_request?
     end
 
     def fetch_generic_object_definition(type, id, data)
-      id ||= data['name']
-      resource_search(id, type, collection_class(type))
+      resource_search(id || data['name'], type)
     end
 
-    def resource_search(id, type, klass)
-      if id.to_s =~ /\A\d+\z/
-        super
-      else
-        go_def = klass.find_by!(:name => id)
-        filter_resource(go_def, type, klass)
-      end
+    def resource_search(id, type, klass = nil, _key_id = nil)
+      super(id, type, klass, id&.integer? ? "id" : "name")
     end
```

Comment:
Wondering if this is clearer...up to you:

```suggestion
    def resource_search(id, type, klass = nil, key_id = nil)
      key_id = id&.integer? ? "id" : "name"
      super
    end
```

Reply:
I link I like the old way but not a hill... done